express/blocks/pricing-hub/pricing-hub.js

```diff
@@ -0,0 +1,489 @@
+    }
+    if (currentUrl.searchParams.has('srcUrl')) {
+      rUrl = replaceUrlParam(rUrl, 'srcUrl', currentUrl.searchParams.get('srcUrl'));
+    }
+  }
+
+  if (currentUrl.searchParams.has('code')) {
+    planUrl.searchParams.set('code', currentUrl.searchParams.get('code'));
+  }
+
+  if (currentUrl.searchParams.get('rUrl')) {
+    rUrl = currentUrl.searchParams.get('rUrl');
+  }
+
+  if (rUrl) planUrl.searchParams.set('rUrl', rUrl.toString());
+  return planUrl.href;
+}
+
+async function fetchPlan(planUrl) {
+  if (!window.pricingPlans) {
+    window.pricingPlans = {};
+  }
+
+  let plan = window.pricingPlans[planUrl];
+
+  if (!plan) {
+    plan = {};
+    const link = new URL(planUrl);
+    const params = link.searchParams;
+
+    plan.url = planUrl;
+    plan.country = 'us';
+    plan.language = 'en';
+    plan.price = '9.99';
+    plan.currency = 'US';
+    plan.symbol = '$';
+
+    if (planUrl.includes('/sp/')) {
+      plan.offerId = 'FREE0';
+      plan.frequency = 'monthly';
+      plan.name = 'Free';
+      plan.stringId = 'free-trial';
+    } else {
+      plan.offerId = params.get('items[0][id]');
+      plan.frequency = null;
+      plan.name = 'Premium';
+      plan.stringId = '3-month-trial';
+    }
+
+    if (plan.offerId === '70C6FDFC57461D5E449597CC8F327CF1' || plan.offerId === 'CFB1B7F391F77D02FE858C43C4A5C64F') {
+      plan.frequency = 'Monthly';
+    } else if (plan.offerId === 'E963185C442F0C5EEB3AE4F4AAB52C24' || plan.offerId === 'BADDACAB87D148A48539B303F3C5FA92') {
+      plan.frequency = 'Annual';
+    } else {
+      plan.frequency = null;
+    }
+
+    const countryOverride = new URLSearchParams(window.location.search).get('country');
+    const offer = await getOffer(plan.offerId, countryOverride);
+
+    if (offer) {
+      plan.currency = offer.currency;
+      plan.price = offer.unitPrice;
+      plan.formatted = `${offer.unitPriceCurrencyFormatted}`;
+      plan.country = offer.country;
+      plan.vatInfo = offer.vatInfo;
+      plan.language = offer.lang;
+      plan.rawPrice = offer.unitPriceCurrencyFormatted.match(/[\d\s,.+]+/g);
+      plan.prefix = offer.prefix ?? '';
+      plan.suffix = offer.suffix ?? '';
+      plan.formatted = plan.formatted.replace(plan.rawPrice[0], `<strong>${plan.prefix}${plan.rawPrice[0]}${plan.suffix}</strong>`);
+
+      if (plan.name !== 'Free') {
+        plan.formatted += '*';
+      }
+    }
+
+    window.pricingPlans[planUrl] = plan;
+  }
+
+  return plan;
+}
+
+function pushBundleAnalytics(plan, bundleType) {
+  const adobeEventName = 'adobe.com:express:pricing:bundleType:selected';
+  const sparkEventName = 'pricing:bundleTypeSelected';
+  const url = new URL(window.location.href);
+  const sparkTouchPoint = url.searchParams.get('touchpointName');
+
+  /* eslint-disable no-underscore-dangle */
+  /* global digitalData _satellite */
+  digitalData._set('primaryEvent.eventInfo.eventName', adobeEventName);
```

Comment:
If digitalData is not set, this will thrown a reference error. Same for _satellite.